update.py

```diff
@@ -61,59 +105,103 @@ def __init__(self, master):
-                self.callback_t = ctypes.CFUNCTYPE(None)	# keep reference
-                self.callback_fn = self.callback_t(shutdown_request)
-                self.updater.win_sparkle_set_shutdown_request_callback(self.callback_fn)
-                self.updater.win_sparkle_init()
-
-            except:
-                print_exc()
-                self.updater = None
-
-        def setAutomaticUpdatesCheck(self, onoroff):
-            if self.updater:
-                self.updater.win_sparkle_set_automatic_check_for_updates(onoroff)
-
-        def checkForUpdates(self):
-            if self.updater:
-                self.updater.win_sparkle_check_update_with_ui()
-
-        def close(self):
-            if self.updater:
-                self.updater.win_sparkle_cleanup()
-            self.updater = None
+                feed = ElementTree.fromstring(r.text)
+            except SyntaxError as ex:
+                print('Syntax error in update_feed file: {}'.format(str(ex)), file=sys.stderr)
+            else:
+
+                for item in feed.findall('channel/item'):
+                    ver = item.find('enclosure').attrib.get('{http://www.andymatuschak.org/xml-namespaces/sparkle}version')
+                    # This will change A.B.C.D to A.B.C+D
+                    sv = semantic_version.Version.coerce(ver)
+
+                    items[sv] = EDMCVersion(version=ver, # sv might have mangled version
+                                            title=item.find('title').text,
+                                            sv=sv
+                    )
+
+                # Look for any remaining version greater than appversion
+                simple_spec = semantic_version.SimpleSpec('>' + appversion)
+                newversion = simple_spec.select(items.keys())
+
+        if newversion:
+            return items[newversion]
+        return None
+
+    def worker(self) -> None:
+        """
+        Thread worker to perform internal update checking and update GUI
+        status if a newer version is found.
+        :return: None
+        """
+        newversion = self.check_appcast()
+
+        if newversion:
+            # TODO: Surely we can do better than this
+            #       nametowidget('.{}.status'.format(appname.lower()))['text']
+            self.root.nametowidget('.{}.status'.format(appname.lower()))['text'] = newversion.title + ' is available'
+            self.root.update_idletasks()
+
+    def close(self) -> None:
+        """
+        Handles the EDMarketConnector.AppWindow.onexit() request.
+
+        NB: We just 'pass' here because:
+         1) We might have a worker() going, but no way to make that
+            co-operative to respond to a "please stop now" message.
+         2) If we're running frozen then we're using (Win)Sparkle to check
+            and *it* might have asked this whole application to quit, in
+            which case we don't want to ask *it* to quit
+
+        :return: None
```

Comment:
I was part way through putting code in to kill the worker and close down (Win)Sparkle when I realised why we shouldn't.  So, documenting it.  May as well be docstring instead of just a comment.